pkg/api/addtask.go

```diff
@@ -0,0 +1,222 @@
+		return
+	}
+	writeJSON(w, map[string]string{
+		"id":      strconv.FormatInt(t.ID, 10),
+		"date":    t.Date,
+		"title":   t.Title,
+		"comment": t.Comment,
+		"repeat":  t.Repeat,
+	})
+}
+
+func editTaskHandler(w http.ResponseWriter, r *http.Request) {
+	var body map[string]any
+	if err := json.NewDecoder(r.Body).Decode(&body); err != nil {
+		writeJSON(w, map[string]any{"error": err.Error()})
+		return
+	}
+	idVal, ok := body["id"]
+	if !ok {
+		writeJSON(w, map[string]any{"error": "missing id"})
+		return
+	}
+	id, err := parseID(idVal)
+	if err != nil {
+		writeJSON(w, map[string]any{"error": "bad id"})
+		return
+	}
+	t := db.Task{
+		ID:      id,
+		Date:    toStr(body["date"]),
+		Title:   strings.TrimSpace(toStr(body["title"])),
+		Comment: strings.TrimSpace(toStr(body["comment"])),
+		Repeat:  strings.TrimSpace(toStr(body["repeat"])),
+	}
+	if t.Title == "" {
+		writeJSON(w, map[string]any{"error": "title is required"})
+		return
+	}
+	if err := fixDate(&t); err != nil {
+		writeJSON(w, map[string]any{"error": err.Error()})
+		return
+	}
+	if err := db.UpdateTask(&t); err != nil {
+		writeJSON(w, map[string]any{"error": err.Error()})
+		return
+	}
+	writeJSON(w, map[string]any{})
+}
+
+func delTaskHandler(w http.ResponseWriter, r *http.Request) {
+	id := strings.TrimSpace(r.URL.Query().Get("id"))
+	if id == "" {
+		writeJSON(w, map[string]any{"error": "missing id"})
+		return
+	}
+	if err := db.DeleteTask(id); err != nil {
+		writeJSON(w, map[string]any{"error": err.Error()})
+		return
+	}
+	writeJSON(w, map[string]any{})
+}
+
+func doneHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		writeJSON(w, map[string]any{"error": "method not allowed"})
+		return
+	}
+	id := strings.TrimSpace(r.URL.Query().Get("id"))
+	if id == "" {
+		writeJSON(w, map[string]any{"error": "missing id"})
+		return
+	}
+	t, err := db.GetTask(id)
+	if err != nil {
+		writeJSON(w, map[string]any{"error": "task not found"})
+		return
+	}
+	if strings.TrimSpace(t.Repeat) == "" {
+		if err := db.DeleteTask(id); err != nil {
+			writeJSON(w, map[string]any{"error": err.Error()})
+			return
+		}
+		writeJSON(w, map[string]any{})
+		return
+	}
+	now := today(time.Now())
+	next, err := NextDate(now, t.Date, t.Repeat)
+	if err != nil {
+		writeJSON(w, map[string]any{"error": fmt.Sprintf("bad repeat: %v", err)})
+		return
+	}
+	if err := db.UpdateDate(next, id); err != nil {
+		writeJSON(w, map[string]any{"error": err.Error()})
+		return
+	}
+	writeJSON(w, map[string]any{})
+}
+
+func fixDate(t *db.Task) error {
+	now := today(time.Now())
+
+	if t.Date == "" {
+		t.Date = now.Format(DateFmt)
+	} else {
+		if _, err := time.Parse(DateFmt, t.Date); err != nil {
+			return errors.New("bad date format")
+		}
+	}
+
+	dt, _ := time.Parse(DateFmt, t.Date)
+
+	if t.Repeat != "" {
+		next, err := NextDate(now, t.Date, t.Repeat)
+		if err != nil {
+			return fmt.Errorf("bad repeat: %v", err)
```

Comment:
При оборачивании нужно использовать `%w` (wrap), а просто при выводе в консоль `%v`
```suggestion
			return fmt.Errorf("bad repeat: %w", err)
```